examples/aligned_accessor/aligned_accessor.cpp

```diff
@@ -0,0 +1,737 @@
+  static_assert(byte_alignment >= sizeof(ElementType),
+		"byte_alignment must be at least sizeof(ElementType).");
+  static constexpr std::size_t overalignment = byte_alignment / sizeof(ElementType);
+  static_assert(overalignment * sizeof(ElementType) == byte_alignment,
+		"overalignment * sizeof(ElementType) must equal byte_alignment.");
+
+  allocation_t<ElementType> allocation{nullptr, delete_raw<ElementType>{}};
+  aligned_pointer_t<ElementType, byte_alignment> aligned_pointer = nullptr;
+  std::size_t num_elements = 0;
+};
+
+template<class ElementType, std::size_t byte_alignment>
+aligned_array_allocation<ElementType, byte_alignment>
+allocate_aligned(std::size_t num_elements)
+{
+  if(num_elements == 0) {
+    return {};
+  }
+  auto allocation = allocate_raw<ElementType, byte_alignment>(num_elements);
+  aligned_pointer_t<ElementType, byte_alignment> aligned_pointer =
+    _MDSPAN_ASSUME_ALIGNED( ElementType, allocation.get(), byte_alignment );
+  return {std::move(allocation), aligned_pointer, num_elements};
+}
+
+// A dynamically allocated array of ElementType,
+// whose zeroth element has byte alignment byte_alignment.
+//
+// ElementType: A trivially copyable type whose size is a power of two bytes.
+//
+// byte_alignment: A power of two that is a multiple of sizeof(ElementType).
+template<class ElementType, std::size_t byte_alignment>
+class aligned_array {
+public:
+  aligned_array(std::size_t num_elements) :
+    alloc(allocate_aligned<ElementType, byte_alignment>(num_elements))
+  {}
+
+  // A pointer to the zeroth element of the array.
+  // It is aligned to byte_alignment bytes.
+  // All num_elements (constructor parameter) elements are valid.
+  aligned_pointer_t<ElementType, byte_alignment>
+  data_aligned() const noexcept
+  {
+    return _MDSPAN_ASSUME_ALIGNED( ElementType, alloc.aligned_pointer, byte_alignment );
+  }
+
+  // A pointer to the first (NOT zeroth) element of the array.
+  // It is aligned only to sizeof(ElementType) bytes.
+  // All num_elements (constructor parameter) elements are valid.
+  //
+  // This exists so that we can compare performance
+  // of aligned and unaligned access.
+  ElementType* data_unaligned() const noexcept
+  {
+    if (alloc.num_elements == 0) {
+      return nullptr;
+    } else { // align only to ElementType
+      return &(alloc.allocation[1]);
+    }
+  }
+
+private:
+  aligned_array_allocation<ElementType, byte_alignment> alloc;
+};
+
+template<class ElementType, std::size_t byte_alignment>
+using aligned_mdspan_1d =
+  stdex::mdspan<ElementType,
+		stdex::dextents<index_type, 1>,
+		stdex::layout_right,
+		aligned_accessor<ElementType, byte_alignment>>;
+
+template<class ElementType>
+using mdspan_1d =
+  stdex::mdspan<ElementType,
+		stdex::dextents<index_type, 1>,
+		stdex::layout_right,
+		stdex::default_accessor<ElementType>>;
+
+#define TICK() const auto tick = std::chrono::steady_clock::now()
+
+using double_seconds = std::chrono::duration<double, std::ratio<1>>;
```

Comment:
Why would you define an alias if you ever only use it in the TOCK macro definition?

Reply:
It makes the definition of `TOCK` shorter and better expresses the code's intent.